Sprint 3/step 2

dex252 opened the pull request.

dex252 committed:
Добавить react-router-dom в проект

dex252 committed:
Подготовить концепцию запросов для выдачи токена

dex252 committed:
Добавить роутинг. Перенести содержимое app в pages/home

dex252 committed:
Пустые страницы и роутинг на них

dex252 committed:
Обновить package.json

dex252 committed:
Вынести header в App.

dex252 committed:
Верстка страницы входа

dex252 committed:
Обновить loader для отображения ошибки в контексте текущего компонента

dex252 committed:
Страница с запросом логина-пароля

dex252 committed:
Страница регистрации

dex252 committed:
Сверстаны страницы восстановления пароля

dex252 committed:
Сверстать страницу Профиль. Правки PR

dex252 committed:
Рефакторинг в запросах login и register

dex252 committed:
Добавил перехватчики для запросов с передачей и обновлением токена. Мелкие правки

dex252 committed:
Логика страницы forgot-password

dex252 committed:
Логика восстановления пароля

dex252 committed:
Подстветка открытой страницы в профиле

dex252 committed:
Убирать сообщение об ошибке на создании заказа после перелогирования для ошибки токена

dex252 committed:
Правки PR

dex252 committed:
Изменение данных пользователя

dex252 committed:
Отмена сохранения пользователя. Правки PR

dex252 committed:
Не чистить заказ после попытки перелогирования

dex252 committed:
Добавить обертку ProtectedRouteElement

dex252 committed:
Выход из профиля

dex252 committed:
Запретить закрытие окна заказа во время выполнения запроса

dex252 committed:
Исправить поведение контроллера с авторизацией

dex252 committed:
Роутинг для reset и forgot password

dex252 committed:
Откатить изменения

dex252 committed:
Решить проблему с доступом к странице reset-password

dex252 committed:
Отобразить имя профиля в шапке

dex252 committed:
Отображение модального окна и полноценного ингредиента

kyzinatra commented on src/components/app/app.jsx:
```diff
@@ -1,69 +1,69 @@
+					<Routes>
+						<Route
+							path='/'
+							element={<Protected>{<HomePage></HomePage>}</Protected>}>
```
Можно лучше: страницы без `children` стоит закрывать сразу `< />`

kyzinatra commented on src/components/app/app.jsx:
```diff
@@ -1,69 +1,69 @@
+						<Route
+							path='/login'
+							element={<Protected onlyUnAuth={true}>{<LoginPage />}</Protected>}
 						/>
```
Можно лучше: можно просто `<Protected onlyUnAuth>`

kyzinatra commented on src/components/router/protected-router.jsx:
```diff
@@ -0,0 +1,41 @@
+
+	//Требуют авторизации и авторизован + не требуют авторизации и не авторизован
+	console.log(`Redirecting any user from ${location.pathname}`);
+	return children;
```
Можно лучше: логи стоит убрать из прод версии

kyzinatra commented on src/components/router/protected-router.jsx:
```diff
@@ -0,0 +1,41 @@
+	}
+
+	//Требуют авторизации и авторизован + не требуют авторизации и не авторизован
+	console.log(`Redirecting any user from ${location.pathname}`);
```
Отлично! Хорошая реализация `ProtectedRouteElement`

kyzinatra commented on src/pages/home/index.jsx:
```diff
@@ -0,0 +1,57 @@
+
+		dispatch(setIngredients());
+		return () => {
+			console.info('UNMOUNT App');
```
По аналогии все лишние логи стоит удалить

kyzinatra commented on src/pages/user/login/index.jsx:
```diff
@@ -0,0 +1,107 @@
+			<Loader loading={loading}>
+				<div style={{ display: 'flex', flexDirection: 'column' }}>
+					<h1 className={`${styles.title} text text_type_main-medium pb-6`}>
+						Вход
```
Надо исправить: все формы должны быть обернуты в тэг `form` и использовать `onSubmit` событие

kyzinatra commented on src/services/store/slices/auth-slice.jsx:
```diff
@@ -0,0 +1,284 @@
+
+			dispatch(authSlice.actions.setUser(user));
+			dispatch(authSlice.actions[_SUCCESS]());
+			return true;
```
Надо исправить: Во всех запросах, которые используют токен, необходимо проверять ошибку `token expires`. В случае, если мы получили эту ошибку, стоит попытаться обновить токен и повторить запрос. 
Для этого можно сделать обертку над `fetch`, к примеру `fetchWithRefresh()`, которая будет этим заниматься. Все это необходимо, чтобы быть точно уверенными в том, что токен даже в случае истечения токен будет обновлен.

kyzinatra commented:
Выглядит очень круто! Видно, что проделана большая работа. Но все же есть моменты на которые стоит обратить внимание.
Подробнее: 

**Что важно исправить:**

-  все формы должны быть обернуты в тэг `form` и использовать `onSubmit` событие

**Можно лучше:**

-  страницы без `children` стоит закрывать сразу `< />`
-  можно просто `<Protected onlyUnAuth>`
-  логи стоит убрать из прод версии


**Что понравилось:**

-  Хорошая реализация `ProtectedRouteElement`


**Обратите внимание, что работа принимается только после исправления всех `«Надо исправить»`.**

dex252 committed:
Удалить лишнюю проверку.с токеном в catch

dex252 committed:
Перенес navigate из метода getOrder в компонент, сделал метод асинхронным с ожиданием

Поддерживает случай, когда токен угнали или изменили пока пользователь находился на странице с заказом

dex252 committed:
Устранение замечаний с передачей параметров в пропсы и сокращением ссылок на страницы

dex252 committed:
Устранить замечание: все формы должны быть обернуты в тэг form и использовать onSubmit событие

dex252 committed:
Закомментировал или удалил большинство логов

Оставил комментарием для дальнейшей разработки

kyzinatra reviewed (approved):
Отличная работа! Принято
Желаю успехов в дальнейшем обучении
